pkg/util/azureclient/azureclient.go

```diff
@@ -70,7 +70,9 @@ func SetupClient(ctx context.Context, log *logrus.Entry, clientName string, clie
 	client.RequestInspector = addAcceptLanguages(languages)
 	client.PollingDelay = 10 * time.Second
 	client.Sender = &retrySender{Sender: client.Sender, log: log, clientName: clientName}
-	//client.Sender = &loggingSender{client.Sender}
+	if os.Getenv("AZURE_DEBUG") != "" {
+		client.Sender = &loggingSender{client.Sender}
```

Comment:
This will be extremely verbose.

Reply:
That's the idea :smile: I want to be able to see the full contents of requests/responses when we have some mystery behaviour like in #1571 where the Azure API was complaining about a request body that was impossible to inspect without making code changes.

Reply:
Note: it will also log the authentication token...  Particularly given such, I think we'd have to find a way to pipe TestConfig all the way down to this level - I don't want us to risk us colliding with an environment variable set on the RP.

Reply:
We could make setupClient part of the struct method and this would allow easier piping.